Tests/NIOTests/EventLoopTest.swift

```diff
@@ -228,4 +228,44 @@ public class EventLoopTest : XCTestCase {
             XCTAssertNoThrow(try group.syncShutdownGracefully())
         #endif
     }
+    
+    public func testCurrentEventLoop() throws {
+        class EventLoopHolder {
+            weak var loop: EventLoop?
+            init(_ loop: EventLoop) {
+                self.loop = loop
+            }
+        }
+        
+        func assertCurrentEventLoop0() throws -> EventLoopHolder {
+            let group = MultiThreadedEventLoopGroup(numThreads: 2)
+            
+            let loop1 = group.next()
+            let currentLoop1 = try loop1.submit {
+                MultiThreadedEventLoopGroup.currentEventLoop
+            }.wait()
+            XCTAssertTrue(loop1 === currentLoop1)
+            
+            let loop2 = group.next()
+            let currentLoop2 = try loop2.submit {
+                MultiThreadedEventLoopGroup.currentEventLoop
+            }.wait()
+            XCTAssertTrue(loop2 === currentLoop2)
+            XCTAssertFalse(loop1 === loop2)
+            
+            let holder = EventLoopHolder(loop2)
+            XCTAssertNotNil(holder.loop)
+            XCTAssertNil(MultiThreadedEventLoopGroup.currentEventLoop)
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+            return holder
+        }
+        
+        let holder = try assertCurrentEventLoop0()
+        
+        // Check we not leak the EventLoop.
+        while holder.loop != nil {
+            // Give the ARC some time to kick in and remove the weak reference.
+            sleep(1)
+        }
```

Comment:
Is this loop actually necessary? ARC should be fully deterministic here if we're using `syncShutdownGracefully`.

Reply:
Without the whole it sometimes fails 🤷‍♂️

Reply:
so sleeping should not fix ARC issues as it's deterministic as @Lukasa says. The problem here is the race between the `MultithreadedEventLoop`'s thread and the main thread running the test. So I think we should fix the comment and then merge.

Reply:
Can we add support for joining `Thread` instead? I'd vastly prefer having a deterministic test than one that has an infinite loop to avoid timing issues.

Reply:
I can do this sure

Reply:
sounds like the right thing to do...